code/modules/customitems/item_defines.dm

```diff
@@ -363,7 +363,9 @@ All custom items with worn sprites must follow the contained sprite system: http
 		item_state = icon_state
 		slot_flags = initial(slot_flags)
 		attack_verb = list("prodded")
-	user.regenerate_icons()
+
+	user.update_inv_l_hand()

```

Comment:
Change this first one to `user.update_inv_l_hand(FALSE)` so that icons aren't updated twice.